lib/kvao/delete-all.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var assert = require('assert');
+var async = require('async');
+var debug = require('debug')('loopback:kvao:delete-all');
+var utils = require('../utils');
+
+/**
+ * Delete all keys (and values) associated to the current model.
+ *
+ * @options {Object} options Unused ATM, placeholder for future options.
+ * @callback {Function} callback
+ * @param {Error} err Error object.
+ * @promise
+ *
+ * @header KVAO.prototype.deleteAll([options, ]cb)
+ */
+module.exports = function deleteAll(options, callback) {
+  if (callback == undefined && typeof options === 'function') {
+    callback = options;
+    options = {};
+  } else if (!options) {
+    options = {};
+  }
+
+  assert(typeof options === 'object', 'options must be an object');
+
+  callback = callback || utils.createPromiseCallback();
+
+  var connector = this.getConnector();
+  if (typeof connector.deleteAll === 'function') {
+    connector.deleteAll(this.modelName, options, callback);
+  } else if (typeof connector.delete === 'function') {
+    debug('falling back to unoptimized `deleteAll`');
+    iterateAndDelete(this.modelName, connector, options, callback);
+  } else {
+    debug('connector does not support `deleteAll`');
+    process.nextTick(function() {
+      var err = new Error('connector does not support `deleteAll`');
+      err.statusCode = 501;
+      callback(err);
+    });
+  }
+  return callback.promise;
+};
+
+function iterateAndDelete(modelName, connector, options, callback) {
```

Comment:
Nitpick: I would personally change the order of arguments from `(modelName, connector,...)` to `(connector, modelName, ...)`, Start with the thing that's changing the least (connector remains the same for the lifetime of a datasource), follow with things that change more often.